OneBusAway/otpkit
# Issue 14 - Add Geolocation support for user's current location

# Add geolocation support for user's current location

## Related Issues
#14 

## Description
This PR related to issue #14. Within this PR, the changes included are:
1. If user opens the map, it will ask a permission to ask for a location
2. When search component in `add favorite location` and `search for location` is clicked, the user's location will appear
3. When users type in search component of add favorite location and search for location, the user's location will dissapear and replaced by searched locations

## Media
https://github.com/OneBusAway/otpkit/assets/74521244/b417b60f-069a-43b0-b476-7d56264b98d1

## Test Instructions

To test this PR, you can do the following:

1. Uninstall the OTPKitDemo app in your simulator
2. Re-run, set the simulator location. For example, [here](https://stackoverflow.com/questions/214416/set-the-location-in-iphone-simulator)
3. See if location permission is asked. Grant the permission
4. Open any origin or destination menu
5. Try to tap on search component. When it's focused or tapped, the row of "My Location" should appear
6. Try to type on the search component, "My Location" row should dissapear

## Additional Context

This PR is rebased from this PR #20. If you want to check for changes for this PR only, you should merge PR first
